### PR TITLE
fix!: ensure `getOperations` returns operation `Transfer Asset` for transfer to contract transactions

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close: https://github.com/FuelLabs/fuels-ts/issues/1394

### CODE REVIEW

File: packages/providers/src/transaction-summary/operations.ts [hunk 1]
```diff
@@ -311,34 +320,74 @@ export function getContractCallOperations({
...
+          ],
+        });
+      }
+    });
+  } else {
+    coinOutputs.forEach((output) => {
+      const input = getInputFromAssetId(inputs, output.assetId);
+      if (input) {
+        const inputAddress = getInputAccountAddress(input);
+        operations = addOperation(operations, {
```
Comment (arboleya) [inline packages/providers/src/transaction-summary/operations.ts:371]: Is it ok to override the `operations` variable on each iteration of this `forEach` loop?

I might be totally out of context, but if only one execution reaches line `371`, maybe doing a `find` before the `forEach` (or breaking the loop when it matches the desired criteria) could be better.

**Review by arboleya — APPROVED** [review]

File: packages/providers/src/transaction-summary/operations.ts (hunk 1 above)
Comment (digorithm) [reply packages/providers/src/transaction-summary/operations.ts:371]: Couldn't there be more than one input in this context?

Comment (Torres-ssf) [reply packages/providers/src/transaction-summary/operations.ts:371]: @arboleya @digorithm good point. 

It is unclear to me if we only expect one operation at this part, or if multiple 'Transfer Asset' operations can happen on the same transaction.

Either way, this logic can be improved.

I've created this issue [here](https://github.com/FuelLabs/fuels-ts/issues/1411) to investigate any potential issues and to improve the code logic being used here

**Comment by blackbeam** [comment]
@Torres-ssf, hi.

I can't get the meaning of the `getTransferOperations` output on [this transaction](https://app.fuel.network/tx/0xc28c74411feaa16861eadbd0505ec000c5af69f4bc018a5c2d4d3f41a844a5ae/simple).
Namely I can't understand why there exists the second entry in the `assetsSent` array:

```typescript
{
  name: 'Transfer asset',
  from: {
    type: 1,
    address: '0x43fea237aa56b945a16acfad737169a4a4e9c13e3c0b683eedd94640434e81d4'
  },
  to: {
    type: 0,
    address: '0xaa3b493e95cbeba71ce48aa85d3113d13de91a25bee095d9ede79c8aa059fe70'
  },
  assetsSent: [
    {
      assetId: '0x0bb7345255ed73ae8f41d00c065c25efd2a7c39fd17ccf6647401a6ccb22650e',
      amount: BN {
        negative: 0,
        words: [ 27311104, 89 ],
        length: 2,
        red: null,
        MAX_U64: '0xFFFFFFFFFFFFFFFF'
      }
    },
    {
      assetId: '0x0000000000000000000000000000000000000000000000000000000000000000',
      amount: BN {
        negative: 0,
        words: [ 27311104, 89 ],
        length: 2,
        red: null,
        MAX_U64: '0xFFFFFFFFFFFFFFFF'
      }
    }
  ]
}
```

**Comment by Torres-ssf** [comment]
> @Torres-ssf, hi.
> 
> I can't get the meaning of the `getTransferOperations` output on [this transaction](https://app.fuel.network/tx/0xc28c74411feaa16861eadbd0505ec000c5af69f4bc018a5c2d4d3f41a844a5ae/simple). Namely I can't understand why there exists the second entry in the `assetsSent` array:
> 
> ```ts
> {
>   name: 'Transfer asset',
>   from: {
>     type: 1,
>     address: '0x43fea237aa56b945a16acfad737169a4a4e9c13e3c0b683eedd94640434e81d4'
>   },
>   to: {
>     type: 0,
>     address: '0xaa3b493e95cbeba71ce48aa85d3113d13de91a25bee095d9ede79c8aa059fe70'
>   },
>   assetsSent: [
>     {
>       assetId: '0x0bb7345255ed73ae8f41d00c065c25efd2a7c39fd17ccf6647401a6ccb22650e',
>       amount: BN {
>         negative: 0,
>         words: [ 27311104, 89 ],
>         length: 2,
>         red: null,
>         MAX_U64: '0xFFFFFFFFFFFFFFFF'
>       }
>     },
>     {
>       assetId: '0x0000000000000000000000000000000000000000000000000000000000000000',
>       amount: BN {
>         negative: 0,
>         words: [ 27311104, 89 ],
>         length: 2,
>         red: null,
>         MAX_U64: '0xFFFFFFFFFFFFFFFF'
>       }
>     }
>   ]
> }
> ```

@blackbeam, most likely this is a [bug](https://github.com/FuelLabs/fuels-ts/issues/1774). I will investigate that.

**Pull request merged**